src/components/table.ts

```diff
@@ -228,39 +228,49 @@ export module Component {
     private _determineGuarantees(offeredWidths: number[], offeredHeights: number[]): _LayoutAllocation {
       var requestedWidths  = _Util.Methods.createFilledArray(0, this._nCols);
       var requestedHeights = _Util.Methods.createFilledArray(0, this._nRows);
-      var layoutWantsWidth  = _Util.Methods.createFilledArray(false, this._nCols);
-      var layoutWantsHeight = _Util.Methods.createFilledArray(false, this._nRows);
+      var columnNeedsWidth  = _Util.Methods.createFilledArray(false, this._nCols);
+      var rowNeedsHeight = _Util.Methods.createFilledArray(false, this._nRows);
+
       this._rows.forEach((row: AbstractComponent[], rowIndex: number) => {
         row.forEach((component: AbstractComponent, colIndex: number) => {
           var spaceRequest: _SpaceRequest;
           if (component != null) {
             spaceRequest = component._requestedSpace(offeredWidths[colIndex], offeredHeights[rowIndex]);
           } else {
-            spaceRequest = {width: 0, height: 0, wantsWidth: false, wantsHeight: false};
+            spaceRequest = {
+              width: 0,
+              height: 0
+            };
           }
 
           var allocatedWidth = Math.min(spaceRequest.width, offeredWidths[colIndex]);
-          var allocatedHeight = Math.min(spaceRequest.height, offeredHeights[rowIndex]);
+          requestedWidths[colIndex] = Math.max(requestedWidths[colIndex], allocatedWidth);
 
-          requestedWidths [colIndex] = Math.max(requestedWidths [colIndex], allocatedWidth );
+          var allocatedHeight = Math.min(spaceRequest.height, offeredHeights[rowIndex]);
           requestedHeights[rowIndex] = Math.max(requestedHeights[rowIndex], allocatedHeight);
-          layoutWantsWidth [colIndex] = layoutWantsWidth [colIndex] || spaceRequest.wantsWidth;
-          layoutWantsHeight[rowIndex] = layoutWantsHeight[rowIndex] || spaceRequest.wantsHeight;
+
+          var componentNeedsWidth = spaceRequest.width > offeredWidths[colIndex];
+          columnNeedsWidth[colIndex] = columnNeedsWidth[colIndex] || componentNeedsWidth;
+
+          var componentNeedsHeight = spaceRequest.height > offeredHeights[rowIndex];
+          rowNeedsHeight[rowIndex] = rowNeedsHeight[rowIndex] || componentNeedsHeight;
```

Comment:
The actual objective is to simplify the return signature of `requestedSpace()`, although the way that is accomplished is to generalize `wantsWidth` and `wantsHeight`.